homeassistant/components/lyric/sensor.py

```diff
@@ -0,0 +1,116 @@
+    lyric = hass.data[DOMAIN][DATA_LYRIC_CLIENT]
+
+    try:
+        devices = lyric.devices()
+    except Exception as exception:
+        raise PlatformNotReady from exception
+
+    hass.data[DOMAIN][DATA_LYRIC_DEVICES] = devices
+
+    devices = []
+    for location, device in lyric.devices():
+        if device.indoorTemperature:
+            devices.append(LyricSensor(
+                device, location, hass, 'indoorTemperature', 'Temperature',
+                hass.config.units.temperature_unit, 'mdi:thermometer'))
+        if device.indoorHumidity:
+            devices.append(LyricSensor(
+                device, location, hass, 'indoorHumidity', 'Humidity',
+                '%', 'mdi:water-percent'))
+        if device.outdoorTemperature:
+            devices.append(LyricSensor(
+                device, location, hass, 'outdoorTemperature',
+                'Temperature Outside',
+                hass.config.units.temperature_unit, 'mdi:thermometer'))
+        if device.displayedOutdoorHumidity:
+            devices.append(LyricSensor(
+                device, location, hass, 'displayedOutdoorHumidity',
+                'Humidity Outside', '%', 'mdi:water-percent'))
+        if device.nextPeriodTime:
+            devices.append(LyricSensor(
+                device, location, hass, 'nextPeriodTime',
+                'Next Period Time', '', 'mdi:clock'))
+        if device.thermostatSetpointStatus:
+            devices.append(LyricSensor(
+                device, location, hass, 'thermostatSetpointStatus',
+                'Status', '', 'mdi:thermostat'))
+
+    async_add_entities(devices, True)
+
+
+class LyricSensor(LyricDeviceEntity):
+    """Representation of a Lyric thermostat."""
+
+    def __init__(self, device, location, hass, key, key_name,
+                 unit_of_measurement=None, icon=None) -> None:
+        """Initialize the sensor."""
+        unique_id = '{}_{}'.format(
+            device.macID, key_name.replace(" ", "_"))
+        self._unit_of_measurement = unit_of_measurement
+        self._state = None
+        self._available = False
+        self.key = key
+
+        name = '{} {}'.format(device.name, key_name)
+
+        super().__init__(device, location, unique_id, name, icon)
+
+    @property
+    def state(self):
+        """Return the state of the entity."""
+        return self._state
+
+    @property
+    def unit_of_measurement(self) -> str:
+        """Return the unit of measurement."""
+        return self._unit_of_measurement
+
+    async def _lyric_update(self) -> None:
+        """Get values from lyric."""
+        if self.device:
+            if self.key == 'thermostatSetpointStatus':
+                status = getattr(self.device, self.key)
+                if status == 'NoHold':
+                    self._state = 'Following Schedule'
+                elif status == 'HoldUntil':
+                    self._state = 'Held until {}'.format(
+                        self.device.nextPeriodTime[:-3])
```

Comment:
What kind of time is this? Is it an absolute utc timestamp?

Reply:
Here is what this returns:

![image](https://user-images.githubusercontent.com/28114703/61577446-31b6c000-aadf-11e9-87e6-3e9e5e45cb76.png)

Reply:
We want all times to be absolute utc. Please make sure.